packages/tooling/fast-tooling-react/src/form/custom-controls/css/color-picker/color-picker.ts

```diff
@@ -0,0 +1,376 @@
+        this.updateUIValues(false);
+
+        this.proxy.setAttribute("type", "colorpicker");
+
+        if (this.autofocus) {
+            DOM.queueUpdate(() => {
+                this.focus();
+            });
+        }
+    }
+
+    /**
+     * Handles the focus event. When the template has focus the color UI will be visable.
+     * @internal
+     */
+    public handleFocus(): void {
+        this.open = true;
+    }
+
+    /**
+     * Handles the blur event. Hides the color UI when the template loses focus.
+     * @internal
+     */
+    public handleBlur(): void {
+        this.open = false;
+    }
+
+    /**
+     * Handles the internal control's `input` event. This event is fired whenever a user types directly into the primary text field.
+     * @internal
+     */
+    public handleTextInput(): void {
+        this.value = this.control.value;
+        if (this.isValideCSSColor(this.value)) {
+            this.currentRGBColor = parseColor(this.value);
+            this.currentHSVColor = rgbToHSV(this.currentRGBColor);
+            this.updateUIValues(false);
+            this.$emit("change");
+        }
+    }
+
+    /**
+     * Handles the mouse down event on the Sat/Val square and Hue and Alpha sliders. Sets the current targeted element and begins listening for mouse move events.
+     * @param param ['sv','h','a'] - string specifying which color property is being modified with the mouse.
+     * @param e - A reference to the mouse event.
+     */
+    public handleMouseDown(param: string, e: MouseEvent) {
+        this.currentMouseTarget = <HTMLElement>e.composedPath()[0];
+        this.currentMouseParam = param;
+        this.updateFromMouseEvent(e.pageX, e.pageY);
+        this.mouseActive = true;
+    }
+
+    /**
+     * Handles the mouse move event within the color UI. Is only called once the mouseActive is set to true.
+     * @param e - Reference to the Mouse Event
+     */
+    public handleMouseMove(e: MouseEvent) {
+        this.updateFromMouseEvent(e.pageX, e.pageY);
+    }
+
+    /**
+     * Handles the mouse up event within the color UI. Disables listening for mouse move events.
+     * @param e - Reference to the Mouse Event
+     */
+    public handleMouseUp(e: MouseEvent) {
+        this.updateFromMouseEvent(e.pageX, e.pageY);
+        this.currentMouseTarget = null;
+        this.currentMouseParam = null;
+        this.mouseActive = false;
+    }
+
+    /**
+     * Handles changes to any of the color property text inputs typed by the user.
+     * @param param ['r','g','b','a','h','s','v'] - String specifying which color value is being modified.
+     * @param e - Reference to the event.
+     */
+    public handleTextValueInput(param: string, e: Event) {
+        const inputVal = (<HTMLInputElement>e.composedPath()[0]).value;
+        if (isNullOrWhiteSpace(inputVal) || Number.isNaN(inputVal)) {
+            return;
+        }
+        let newVal: number = parseInt(inputVal);
```

Comment:
It's safest to provide a radix to `parseInt` see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#parameters

```suggestion
        let newVal: number = parseInt(inputVal, 10);
```

Reply:
Done.